yang/ietf-schedule.yang

```diff
@@ -540,4 +555,71 @@ module ietf-schedule {
         "Defines a list of exceptions for recurrence.";
     }
   }
+  
+  grouping schedule-status {
+    description
+      "This grouping is defined to identify common properties of 
+       scheduling status.";
+    leaf schedule-id {
+      type string;
+      description
+        "The schedule identifier that globally identifies a schedule.";
+    }
+    leaf schedule-type {
+      type identityref {
+        base schedule-type;
+      }
+      description
+        "The schedule type";
+    }    
+    leaf last-update {
+      type yang:date-and-time;
+      description
+        "The timestamp that the schedule is last updated.";
+    }
+    leaf time-zone-identifier {
+      type sys:timezone-name;
+      description
+        "Indicates the referenced time zone identifier of the schedule.";
+    }
+    leaf version {
+      type uint16;
+      description
+        "The version number of the schedule.";
+    }
+    leaf state {
+      type enumeration {
+        enum enabled {
+          value 1;
+          description
+            "The schedule is currently enabled.";
+        }
+        enum disabled {
+          value 2;
+          description
+            "The schedule is currently disabled.";
+        }        
+        enum out-of-date {
+          value 3;
+          description
+            "The schedule is currently out-of-date.";
+        }
+      }
+      description
+        "The current state of the schedule.";
+    }
+    leaf-list occurrence-list {
```

Comment:
Shouldn't this be always ro?

Also, maintaining an indefinite list of occurrences may lead to memory overflow.

Rather, I think we can maintain a counter of occurrences since the sch was enabled + a separate attribute for the last-occurrence timestamp. No?

Reply:
> Shouldn't this be always ro?
yes, i think this as ro.  and the update of schedule can be done by modules that use period and recurrence groupings.

> Also, maintaining an indefinite list of occurrences may lead to memory overflow.
you are right. so i guess it cannot be defined this way.

> Rather, I think we can maintain a counter of occurrences since the sch was enabled + a separate attribute for the last-occurrence timestamp. No?
works for me.